### PR TITLE
Implemented HubSpot Cog

### PR DESCRIPTION
**Pull request by amcpanaligan**
# Notes
1. Implemented ContactFieldEqualsStep
2. The auth fields are temporarily set to accept `apiKey` only as it is in `atomatest`
3. Tests are not yet created. I will update this pull request from time to time.

^ Updates
1. implemented `createorupdate-validate-delete` steps
2. Added a `scenario file` to test the three
3. Unit tests for `createorupdate-validate-delete` steps at 100%

### CODE REVIEW

File: test/scenarios/hubspot-contact-scenario.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+steps:
+- step: create or update a hubspot contact
+  data:
+    contact:
+        email: hubspot@sample.com
+        lastname: Doe
+        company: Sample Company
+- step: the company field on hubspot contact hubspot@sample.com should be Sample Company
+- step: the lastname field on hubspot contact hubspot@sample.com should be Doe
+- step: delete the hubspot@sample.com hubspot contact
```
Comment (iamEAP) [inline test/scenarios/hubspot-contact-scenario.yml]: - [ ] Nit: Newline at end of file

File: src/steps/contact-field-equals.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    const stepData: any = step.getData() ? step.getData().toJavaScript() : {};
+    const expectation = stepData.expectation;
+    const email = stepData.email;
+    const field = stepData.field;
+
+    try {
+      const contact = await this.client.getContactByEmail(email);
+
+      if (!contact.properties[field]) {
+        return this.error('Expected field %s was invalid.', [
```
Comment (iamEAP) [inline src/steps/contact-field-equals.ts]: - [ ] Suggested error message: `Couldn't check field %s on contact: field doesn't exist.

File: src/steps/contact-delete.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        return this.pass('Successfully deleted contact %s', [
+          email,
+        ]);
+      } else {
+        return this.fail('Unable to delete contact. Reason: %s', [
+          data.reason,
+        ]);
+      }
+    } catch (e) {
+      return this.error('There was an error delete contact in HubSpot: %s', [
```
Comment (iamEAP) [inline src/steps/contact-delete.ts]: - [ ] Suggested error message: `There was an error deleting the HubSpot contact: %s`

File: src/steps/contact-delete.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    try {
+      const data = await this.client.deleteContactByEmail(email);
+
+      if (data.deleted) {
+        return this.pass('Successfully deleted contact %s', [
+          email,
+        ]);
+      } else {
+        return this.fail('Unable to delete contact. Reason: %s', [
```
Comment (iamEAP) [inline src/steps/contact-delete.ts]: - [ ] Suggested failure message: `Unable to delete contact: %s`

File: src/steps/contact-create-or-update.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+      if (data) {
+        return this.pass('Successfully created or updated contact %s', [
+          email,
+        ]);
+      } else {
+        return this.fail('Unable to create or update contact');
+      }
+    } catch (e) {
+      return this.error('There was an error creating or updating contacts in HubSpot: %s', [
```
Comment (iamEAP) [inline src/steps/contact-create-or-update.ts]: - [ ] Suggested error message: `There was an error creating or updating the contact in HubSpot: %s`

File: src/steps/contact-delete.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
+/*tslint:disable:no-else-after-return*/
+
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+export class DeleteContactStep extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Delete a hubspot contact';
```
Comment (iamEAP) [inline src/steps/contact-delete.ts]: - [ ] Recommend brand case on `HubSpot` here

File: src/steps/contact-create-or-update.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
+/*tslint:disable:no-else-after-return*/
+
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+export class CreateOrUpdateContactStep extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Create or update a hubspot contact';
```
Comment (iamEAP) [inline src/steps/contact-create-or-update.ts]: - [ ] Recommend brand case on `HubSpot`, here.

**Comment by iamEAP** [comment]
Merging this in as-is--I will handle the very minor edits above in a follow-up.  Thanks again @amcpanaligan!

**Comment by amcpanaligan** [comment]
Then, thanks a lot! Apologies

**Pull request merged**